### ISSUE TITLE
Deleting Backwards a Line & Markdown Links

### ISSUE THREAD

**Issue by keithadams** [tadashi-aikawa/obsidian-various-complements-plugin#65]
When deleting backwards a line, Various Complements has weird behaviors. 3 things to notice after deleting a trailing line:

1. Various Complements offers to complete the already complete word
2. The cursor is not at the end of the above line. It's a few characters back
3. The markdown link is destroyed if you execute the completion

**GIF:**
![various complements](https://user-images.githubusercontent.com/11220719/148121039-c51a6e46-fcf3-4315-9a2f-99851c422e6e.gif)

```
LINK BEFORE
[Various Complements](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin)

LINK AFTER
[Various Complements](https://github.com/tadashi-aikawa/obsidian-various-complemeComplements
```

**Comment by tadashi-aikawa** [comment]
Hi, @keithadams.

I couldn't reproduce it. Does it reproduce in the case that uses default theme and no plugins?

**Comment by Aaron-Bopp** [comment]
I am also running into this issue. It seems to happen with any special character. I tried switching settings around and could not get it to not occur. Maybe this snippet will help `|` is the cursor
```
testing
testing.|
```
When typing the second testing, it will suggest the completion because of the previous lines, but the suggestion doesn't go away after I type the period. If I accept the suggestion, then it becomes `ttesting`

**Comment by keithadams** [comment]
I tried with a new vault, only VC plugin installed. Same result:

![various complements delete line](https://user-images.githubusercontent.com/11220719/148287069-c04b8a96-1e3e-4e06-9809-a8cc0104b6ab.gif)

I'm using v0.13.18. Problem happens in legacy editor and new editor. Also, happens in Live Preview and Source View.

**Comment by keithadams** [comment]
Although, I should mention the cursor problem is gone with no plugins. I suspect the plugin I use to automatically insert favicons is the problem for that.

**Comment by tadashi-aikawa** [comment]
@keithadams @Aaron-Bopp 
Thank you for details! I could reproduce it 👍

**Comment by tadashi-aikawa** [comment]
@keithadams @Aaron-Bopp 
I have released to [v4.0.1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/4.0.1) 🚀

Please confirm whether it works as you expected.

**Comment by keithadams** [comment]
Yes, it's fixed. Thanks!
